### DISCUSSION TITLE
[建议] 固定下载进度条

### DISCUSSION THREAD

**Discussion by zjns** [yutto-dev/yutto#37] [General]
现在的进度条不停的跑，Terminal 里占用一大片，不太友好，希望能固定进度条展示，没必要把每时刻的进度都打印到 Terminal。可以参考这个项目的实现 https://github.com/nathom/streamrip ，也是 Python 项目。

**Comment by SigureMo** [comment]
我不太明白不停的跑是什么意思，会不会是你的 Terminal 宽度太窄导致进度条无法完整显示？

正常来说进度条是这样的：

![image](https://user-images.githubusercontent.com/38436475/147409380-80e76390-a838-4947-b715-6eda1ab6ab54.png)

它是固定在底部作为 StatusBar 展示的，只会占一行……

另外希望能够提供更多信息，比如截图……

**Reply by zjns** [reply]
晕，原来我这个是个例，确实是窗口不够宽的问题。
设备是 macOS，用的自带的 Terminal，窗口大小是默认的 80x24，字体大小是 15，调到 90 就没问题了。

但我觉得应该按照窗口的大小来决定进度条的长度比较好，[streamrip](https://github.com/nathom/streamrip) 这个项目就做的比较好，并且如果在手机上使用，屏幕本来就比较窄，很容易就出现我这种情况。
@SigureMo

**Reply by SigureMo** [reply]
它们固定占宽 10 字符宽度……`6.46 GiB`前面会补空格

**Reply by SigureMo** [reply]
![image](https://user-images.githubusercontent.com/38436475/147411837-3383b29d-ff14-48a9-8f14-80fc3e523832.png)

![image](https://user-images.githubusercontent.com/38436475/147411842-b32edf1b-d064-4ce2-a6cc-e0e2fedd3806.png)

![image](https://user-images.githubusercontent.com/38436475/147411847-2707cc3e-bc1f-4370-afff-b1a6782958f0.png)

不就是自动调整宽度嘛……早就在 1.x 做过了……粘过来了……

**Comment by SigureMo** [comment]
已发布 2.0.0-beta.7

**Reply by zjns** [reply]
大哥神速！🐂🍺

**Comment by SigureMo** [comment]
另外截图的 TitleBar 上显示的命令暴露 SESSDATA 了，我帮你把这个回复删掉了，记得保护好隐私

删掉的回复：

![image](https://user-images.githubusercontent.com/38436475/147412326-3d95c943-d396-4071-8ec0-dd2e930af8dc.png)

**Reply by zjns** [reply]
多谢

**Comment by zjns** [comment]
@SigureMo 在手机上还是有问题...
![Termux](https://user-images.githubusercontent.com/10793522/147427730-35dcfc6f-58b1-4552-99fd-1c3e262cf5af.png)

**Reply by SigureMo** [reply]
手机上实在是太窄了……现在进度条最低 15 字符……再窄进度条看不见了都……这样吧，我之后设置进度条最窄 10 字符，你的手机可能刚好能放得下也可能刚好放不下。然后我再加一个逻辑，低于 10 字符的进度条不进行展示，仅展示数字，你觉得怎么样？因为太窄的进度条实在没啥意义了。

**Reply by zjns** [reply]
> 手机上实在是太窄了……现在进度条最低 15 字符……再窄进度条看不见了都……这样吧，我之后设置进度条最窄 10 字符，你的手机可能刚好能放得下也可能刚好放不下。然后我再加一个逻辑，低于 10 字符的进度条不进行展示，仅展示数字，你觉得怎么样？因为太窄的进度条实在没啥意义了。

那就最窄10个字符吧，其实把页面再缩小点也能放得下。没必要太纠结这个，就这样吧。

**Reply by SigureMo** [reply]
嗯，我之后做一下，但是这次不会尽快发版了，要等一段时间再发布新版本，如果需要的话可以尝试手动通过 git clone 方式安装

**Comment by xincheng213618** [comment]
进度条可以用之前的那种刷新模式吗，windows cmd 上一直乱刷

**Reply by SigureMo** [reply]
不应该吧……cmd 上应该没有这个问题吧？如果是颜色显示问题的话试试加上 `--no-color`，不过进度条可能看不清了，不过我还是建议使用 Windows Terminal

**Comment by xincheng213618** [comment]
[](url
![7CNS5`B{)8%VO 079BWF$6U](https://user-images.githubusercontent.com/25397726/147455415-20f51409-2779-40c1-a8e2-cbb4fcae42f4.png)
)

**Reply by SigureMo** [reply]
嗯，是颜色的问题，使用 Windows Terminal 应该可以解决，如果一定要用 cmd 的话，加上 `--no-color`

**Reply by xincheng213618** [reply]
好的，我试试
